ibis/expr/operations/relations.py

```diff
@@ -439,42 +439,13 @@ def schema(self):
     def blocks(self):
         return bool(self.selections)
 
+    @util.deprecated(instead="instantiate Selection directly", version="4.0.0")
     def substitute_table(self, table_expr):
         return Selection(table_expr, self.selections)
 
     def root_tables(self):
         return [self]
 
-    def can_add_filters(self, wrapped_expr, predicates):
```

Comment:
Should I restore these as well?

Reply:
Yeah I would keep everything that isn't prefixed with an underscore and mark it deprecated. We can delete them in the 4.0.0 release.